src/main/kotlin/com/sourcegraph/cody/chat/ui/LLMModelDropdown.kt

```diff
@@ -0,0 +1,73 @@
+package com.sourcegraph.cody.chat.ui
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.invokeLater
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.ComboBox
+import com.intellij.ui.MutableCollectionComboBoxModel
+import com.intellij.util.concurrency.annotations.RequiresEdt
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.cody.agent.protocol.ChatModelsParams
+import com.sourcegraph.cody.agent.protocol.ChatModelsResponse
+import com.sourcegraph.cody.chat.SessionId
+import com.sourcegraph.cody.config.CodyAccount.Companion.isEnterpriseAccount
+import com.sourcegraph.cody.config.CodyAuthenticationManager
+import com.sourcegraph.cody.ui.ChatModel
+import com.sourcegraph.cody.ui.LLMModelComboBoxItem
+import com.sourcegraph.cody.ui.LLMModelComboBoxRenderer
+import com.sourcegraph.common.BrowserOpener
+import java.util.concurrent.TimeUnit
+
+class LLMModelDropdown(val project: Project, initiallySelected: ChatModel?) :
+    ComboBox<LLMModelComboBoxItem>(MutableCollectionComboBoxModel()) {
+
+  init {
+    val activeAccountType = CodyAuthenticationManager.instance.getActiveAccount(project)
+    if (activeAccountType.isEnterpriseAccount()) {
+      isEnabled = false
+    }
+
+    renderer = LLMModelComboBoxRenderer()
+    if (initiallySelected != null) {
+      addItem(LLMModelComboBoxItem(initiallySelected.icon, initiallySelected.displayName))
+    }
+  }
+
+  fun fetchAndUpdateModels(sessionId: SessionId) {
+    CodyAgentService.withAgent(project) { agent ->
+      val chatModels = agent.server.chatModels(ChatModelsParams(sessionId))
+      val response = chatModels.completeOnTimeout(null, 4, TimeUnit.SECONDS).get()
+      ApplicationManager.getApplication().invokeLater { updateModels(response.models) }
+    }
+  }
+
+  @RequiresEdt
+  private fun updateModels(models: List<ChatModelsResponse.ChatModelProvider>) {
+    removeAllItems()
+    models.forEach { provider ->
+      val model = ChatModel.fromAgentName(provider.model)
+      val name = if (model == ChatModel.UNKNOWN_MODEL) provider.model else model.displayName
```

Comment:
I kind of don't like that `ChatModel.UNKNOWN_MODEL` from the first time I have seen it.
What does it mean it's unknown?
For us doesn't it always mean that it is default? And if so, maybe we should name it `DEFAULT_MODEL`

Reply:
I agree with your feelings about `ChatModel.UNKNOWN_MODEL` and the enum approach overall. 

I'd rather not make any changes to the current approach in this PR. Let's try to limit these changes to CTA. That particular snippet has been moved from the chat panel (this is more like a part of the refactor, not my actual changes). I will more more than happy to reimplement the model handling in a following PR.